core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeResponse.java

```diff
@@ -141,15 +177,21 @@ public void readFrom(StreamInput in) throws IOException {
         for (int i = 0; i < size; i++) {
             tokens.add(AnalyzeToken.readAnalyzeToken(in));
         }
+        detail = in.readOptionalStreamable(DetailAnalyzeResponse::new);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeVInt(tokens.size());
-        for (AnalyzeToken token : tokens) {
-            token.writeTo(out);
+        if (tokens != null) {
+            out.writeVInt(tokens.size());
```

Comment:
I think this won't work. You are reading an int in readFrom() but you are optionally writing an int.